pkg/compute/deploy.go

```diff
@@ -41,13 +56,47 @@ func NewDeployCommand(parent common.Registerer, client api.HTTPClient, globals *
 	c.CmdClause.Flag("service-id", "Service ID").Short('s').StringVar(&c.manifest.Flag.ServiceID)
 	c.CmdClause.Flag("version", "Number of version to activate").Action(c.Version.Set).IntVar(&c.Version.Value)
 	c.CmdClause.Flag("path", "Path to package").Short('p').StringVar(&c.Path)
+	c.CmdClause.Flag("domain", "The name of the domain associated to the package").StringVar(&c.Domain)
+	c.CmdClause.Flag("backend", "A hostname, IPv4, or IPv6 address for the package backend").StringVar(&c.Backend)
+	c.CmdClause.Flag("backend-port", "A port number for the package backend").UintVar(&c.BackendPort)
 
 	return &c
 }
 
 // Exec implements the command interface.
 func (c *DeployCommand) Exec(in io.Reader, out io.Writer) (err error) {
-	var progress text.Progress
+	// Exit early if no token configured.
+	_, s := c.Globals.Token()
+	if s == config.SourceUndefined {
+		return errors.ErrNoToken
+	}
+
+	var (
+		domain, backend string
+		backendPort     uint
+	)
+
+	serviceID, sidSrc := c.manifest.ServiceID()
+	if sidSrc == manifest.SourceUndefined {
+		text.Output(out, "You don't currently have a service defined. This utility will walk you through defining the resources necessary to create a Compute@Edge service.")
```

Comment:
```suggestion
		text.Output(out, "There is no Fastly service associated with this package. To connect to an existing service add the Service ID to the fastly.toml file or otherwise press enter to create one now.")
```